gadget/gadget.go

```diff
@@ -1008,11 +1111,16 @@ func validateCrossVolumeStructure(structures []VolumeStructure, knownStructures
 			}
 		}
 
-		if *(ps.Offset) < previousEnd {
-			previous := structures[pidx-1]
-			return fmt.Errorf("structure %q overlaps with the preceding structure %q", ps.Name, previous.Name)
+		// We are assuming ordered structures
+		if ps.Offset != nil {
+			if *(ps.Offset) < previousEnd {
+				return fmt.Errorf("structure %q overlaps with the preceding structure %q", ps.Name, structures[pidx-1].Name)
+			}
+			previousEnd = *(ps.Offset) + quantity.Offset(ps.MinSize)
+		} else {
+			previousEnd += quantity.Offset(ps.MinSize)
```

Comment:
as this function is about self-consistency of a gadget it's a bit unclear why it's using min-size vs size, if the gadget is used on installation size will be used, not min-size

Reply:
Good point, I used originally MinSize as I was thinking in the minimal restrictions for the volume to be valid, but certainly it is the other way around and any possible size specified should be compatible with the volume definitions. Changed to using Size.